cmd/dogstatsd/main_windows.go

```diff
@@ -6,19 +6,370 @@
+			}
+			if val, _, err = k.GetStringValue("proxy_port"); err == nil && val != "" {
+				u.Host = u.Host + ":" + val
+			}
+			if user, _, _ := k.GetStringValue("proxy_user"); err == nil && user != "" {
+				if pass, _, _ := k.GetStringValue("proxy_password"); err == nil && pass != "" {
+					u.User = url.UserPassword(user, pass)
+				} else {
+					u.User = url.User(user)
+				}
+			}
+		}
+		proxyMap := make(map[string]string)
+		proxyMap["http"] = u.String()
+		proxyMap["https"] = u.String()
+		overrides["proxy"] = proxyMap
+	} else {
+		log.Debug("proxy key not found, not setting proxy config")
+	}
+	if val, _, err = k.GetStringValue("site"); err == nil && val != "" {
+		overrides["site"] = val
+		log.Debugf("Setting site to %s", val)
+	}
+	if val, _, err = k.GetStringValue("dd_url"); err == nil && val != "" {
+		overrides["dd_url"] = val
+		log.Debugf("Setting dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("logs_dd_url"); err == nil && val != "" {
+		overrides["logs_config.logs_dd_url"] = val
+		log.Debugf("Setting logs_config.dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("process_dd_url"); err == nil && val != "" {
+		overrides["process_config.process_dd_url"] = val
+		log.Debugf("Setting process_config.process_dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("trace_dd_url"); err == nil && val != "" {
+		overrides["apm_config.apm_dd_url"] = val
+		log.Debugf("Setting apm_config.apm_dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("py_version"); err == nil && val != "" {
+		overrides["python_version"] = val
+		log.Debugf("Setting python version to %s", val)
+	}
+
+	// apply overrides to the config
+	config.AddOverrides(overrides)
+
+	// build the global agent configuration
+	err = setupConfigWithoutSecrets("")
+	if err != nil {
+		return fmt.Errorf("unable to set up global agent configuration: %v", err)
+	}
+
+	// dump the current configuration to datadog.yaml
+	b, err := yaml.Marshal(config.Datadog.AllSettings())
+	if err != nil {
+		log.Errorf("unable to unmarshal config to YAML: %v", err)
+		return fmt.Errorf("unable to unmarshal config to YAML: %v", err)
+	}
+	// file permissions will be used only to create the file if doesn't exist,
+	// please note on Windows such permissions have no effect.
+	if err = ioutil.WriteFile(datadogYamlPath, b, 0640); err != nil {
+		log.Errorf("unable to unmarshal config to %s: %v", datadogYamlPath, err)
+		return fmt.Errorf("unable to unmarshal config to %s: %v", datadogYamlPath, err)
+	}
+
+	valuenames := []string{"api_key", "tags", "hostname",
+		"proxy_host", "proxy_port", "proxy_user", "proxy_password", "cmd_port"}
+	for _, valuename := range valuenames {
+		k.DeleteValue(valuename)
+	}
+	for valuename := range subServices {
+		k.DeleteValue(valuename)
+	}
+	log.Debugf("Successfully wrote the config into %s\n", datadogYamlPath)
+
+	return nil
+}
+
+// SetupConfigWithoutSecrets fires up the configuration system without secrets support
```

Comment:
The next two functions are almost identical to the present code here: https://github.com/DataDog/datadog-agent/blob/master/cmd/agent/common/helpers.go#L21-L48

If we could avoid repeating this, that would be great.